samples/concepts.js

```diff
@@ -1018,11 +1018,11 @@ class Transaction extends TestHelper {
       // Restore `datastore` to the mock API.
       datastore = datastoreMock;
       assert.strictEqual(
```

Comment:
could we also add a `assert.equal` test that demonstrates `valueOf` gets called automatically? will it not get called in a `strictEqual` check?

Reply:
added additional checks. resolving.